src/components/PaginationFooter/context/PaginationProvider.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { createContext, useContext, useMemo } from 'react';
+import type { stardust } from '@nebula.js/stardust';
+import type { FooterStyle, PaginationContentProps } from '../types';
+import getFooterStyle from '../utils';
+import useTranslations from '../../../hooks/useTranslations';
+
+interface IPaginationProvider extends PaginationContentProps {
+  footerStyle: FooterStyle;
+  tabIndex: number;
+  width: number;
+}
+
+interface PaginationProviderProps extends PaginationContentProps {
+  children: React.ReactNode;
+}
+
+export const areTabStopsEnabled = (keyboard: stardust.Keyboard) => !keyboard.enabled || keyboard.active;
+
+const NOOP = {} as IPaginationProvider;
+
+const PaginationProviderContext = createContext<IPaginationProvider>(NOOP);
+
+export const usePaginationContext = (): IPaginationProvider => useContext(PaginationProviderContext);
+
+const PaginationProvider = ({
+  children,
+  direction,
+  pageInfo,
+  handleChangeRowsPerPage,
+  footerContainer,
+  isSelectionMode,
+  handleChangePage,
+  totalRowCount,
+  totalColumnCount,
+  totalPages,
```

Comment:
I saw this part in PageOptions.tsx:
```jsx
const pageOptionsLength = Math.min(totalPages, MAX_PAGE_OPTIONS_LENGTH);
const Options = Array.from(Array(pageOptionsLength).keys()).map(...)
```

and it got me thinking what if `pageOptionsLength` which relies on `totalPages` where result in `NaN`. Since it's generated at run time, that would mean any non-correct value (NaN) the hosts of this component (PVT, SNT) may provide to this will result in a failure in this component. 

I think it would be better to consider a default value for this as 0, or more default values for others potential passed props that is generated in run time

Reply:
From a type prespective you are correct. `NaN` would be a valid type of `number`. But in general I don't think we have all that much handling of corner cases when a number is `NaN`. Most of the places where numbers are consumed in both `sn-table` and `sn-pivot-table` just don't the checks for `NaN`.

If any number coming in to the `PaginationFooter` component is `NaN`. They will be little point in rendering it at all. So I can do a check for `NaN` in `PageInfo`, `rect`, `totalPages`, `totalRowCount` and `totalColumnCount`. If any of those are `NaN` the componet will not render?

Reply:
_"If any number coming in to the PaginationFooter component is NaN, They will be little point in rendering it at all."_ 

I think we will not reach into that step of rendering only dots since this line will throw a runtime error: 
`const Options = Array.from(Array(pageOptionsLength).keys()).map(...)` 
because `Array()` construct only expects number (i think):

<img width="345" alt="Screenshot 2023-09-20 at 12 25 55" src="https://github.com/qlik-oss/nebula-table-utils/assets/29652890/09637dfd-8840-47e2-a52b-c12bafd5352b">


maybe it's a bit overthinking of me and you are right about we trust the typescript on this about always getting number value 😅

Reply:
I think it's a valid concern.

My point was just that if we get an `NaN`, the whole Pagination footer will likely break. We can fix that by simple checking all numbers before rendering anything. But if there is an `NaN` value, the tables will probably break themself because they do not typically check for `NaN`.